### PR TITLE
allow enabling RBOOT_RTC

### PR DESCRIPTION
**Pull request by patrickjahns**
See #715 

@raburton 

I was not sure how to proceed with the other options I mentioned - if it is okay to also allow them via makefile-user.mk please let me know and I will extend the PR

### CODE REVIEW

File: Sming/rboot/Makefile [hunk 1]
```diff
@@ -37,6 +37,9 @@ endif
 ifeq ($(RBOOT_INTEGRATION),1)
 	CFLAGS += -DRBOOT_INTEGRATION
 endif
+ifeq ($(RBOOT_RTC_ENABLED),1)
```
Comment (raburton) [inline Sming/rboot/Makefile]: rBoot is an external component included in sming unmodified, please supply a patch upstream to rBoot and I'll get it merged

File: Sming/Makefile-rboot.mk [hunk 2]
```diff
@@ -264,6 +264,7 @@ RBOOT_FW_BASE := $(abspath $(FW_BASE))
 export RBOOT_BIG_FLASH
 export RBOOT_BUILD_BASE
 export RBOOT_FW_BASE
+export RBOOT_RTC_ENABLED
```
Comment (raburton) [inline Sming/Makefile-rboot.mk]: you might want to add a use of this (commented out by default) to the rBoot sample app makefile with the other options, so people will know it's there

**Comment by patrickjahns** [comment]
I will move the required changes upstream and see how I can integrate it with the examples.

**Comment by patrickjahns** [comment]
- Moved rboot related changes upstream
- included the sample from @slaff for using temp_rom (hope that is okay with you?)
- allowed for the other options to be set (gpio/irom_chksum/config_chksum)

File: samples/Basic_rBoot/app/application.cpp [hunk 3]
```diff
@@ -163,8 +217,21 @@ void init() {
 	Serial.begin(SERIAL_BAUD_RATE); // 115200 by default
 	Serial.systemDebugOutput(true); // Debug output to serial
 	
+	uint8 slot = 0;
+
+#ifdef BOOT_RTC_ENABLED
+	debugf("Temp Mode is enabled");
```
Comment (raburton) [inline samples/Basic_rBoot/app/application.cpp]: rtc being enabled doesn't necessarily mean you are in temp boot mode
also, currently running rom is printed further down this method already...

File: samples/Basic_rBoot/app/application.cpp [hunk 4]
```diff
@@ -193,4 +260,4 @@ void init() {
 	Serial.println();
 	
 	Serial.setCallback(serialCallBack);
-}
+}
```
Comment (raburton) [inline samples/Basic_rBoot/app/application.cpp]: please leave the training newline

File: Sming/Makefile-rboot.mk [hunk 5]
```diff
@@ -12,6 +12,9 @@
 RBOOT_BIG_FLASH  ?= 1
 RBOOT_TWO_ROMS   ?= 0
 RBOOT_RTC_ENABLED ?= 0
+RBOOT_GPIO_ENABLED ?= 0
+RBOOT_IROM_CHKSUM ?= 0
```
Comment (raburton) [inline Sming/Makefile-rboot.mk]: enabling this requires an extra option on the call to esptool2 that creates the rom image (to include the data in the checksum)

Comment (patrickjahns) [inline Sming/Makefile-rboot.mk]: Can you point me towards the right direction here, then I will update the example accordingly

Comment (raburton) [inline Sming/Makefile-rboot.mk]: You need to add `-iromchksum` to the parameters passed to esptool2 when the application rom is created if RBOOT_IROM_CHKSUM is enabled (set to 1). Probably want something like this in the makefile:
```
ifeq ($(RBOOT_IROM_CHKSUM), 1)
  RBOOT_E2_USER_ARGS += -iromchksum
endif
```

**Pull request closed without merging**